src/encoding_implementations/gaussian_cdf.jl

```diff
@@ -54,32 +75,91 @@ julia> decode(encoding, 3)
  0.4
  0.6
 ```
+
+One can also encode the entire vector as an integer.
+
+```jldoctest
+julia> using ComplexityMeasures, Statistics
+
+julia> x = [0.1, 0.4, 0.7, -2.1, 8.0];
+
+julia> μ, σ = mean(x), std(x); encoding = GaussianCDFEncoding(x; μ, σ, c = 2)
+GaussianCDFEncoding(m=5; c=2, μ=1.42, σ=3.840182287340016)
+
+julia> symbol = encode(encoding, x)
+17
+
+julia> decode(encoding, symbol)
+5-element Vector{SVector{1, Float64}}:
+ [0.0]
+ [0.0]
+ [0.0]
+ [0.0]
+ [0.5000000000000001]
+```
 """
-struct GaussianCDFEncoding{T} <: Encoding
+struct GaussianCDFEncoding{m, T, L <: LinearIndices, C <: CartesianIndices, R} <: Encoding
+    m::Int
     c::Int
     σ::T
     μ::T
-    # We require the input data, because we need σ and μ for encoding single values.
-    function GaussianCDFEncoding(; μ::T, σ::T, c::Int = 3) where T
-        new{T}(c, σ, μ)
+
+    # internal fields: LinearIndices/CartesianIndices for encodings/decodings. binencoder
+    # for discretizing the interval [0, 1]
+    linear_indices::L
+    cartesian_indices::C
+    binencoder::R # RectangularBinEncoding
+
+    # The input `m` restricts what length the input scalar/vector can be.
+    function GaussianCDFEncoding(m::Int = 1; μ::T, σ::T, c::Int = 3) where T
```

Comment:
Just FYI, `GaussianCDFEncoding` is created now in a type unstable manner, as the input value `m` decides the type `m` later. This means that if code constructs and calls it in a single function, that is type unstable code.

There is no way to fix this while still allowing `m` as an input.

I think this performance defecit doesn't matter much unless you are looping over calls to `encode(GaussianCDF, χ)`.

Reply:
> I think this performance defecit doesn't matter much unless you are looping over calls to encode(GaussianCDF, χ).

Then we probably should resolve this here, because that is precisely the intention of allowing vector-valued inputs ( upstream).

Reply:
You can resolve this by making the constructor `GaussianCDFEncoding{m}(; μ...)` So that `m` is given directly as a type parameter by the user.

Reply:
@kahaaga you haven't addressed this comment. You should make this change and then merge. You should also create the default constructor `GaussianCDFEncoding(; μ ...)` that assumes `m=1`.